```
[ERROR] /home/travis/build/hawkular/hawkular-agent/hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/cmd/ExportJdrCommand.java:[70,44] cannot find symbol
  symbol:   method isDeleteImmediately()
  location: variable request of type org.hawkular.cmdgw.api.ExportJdrRequest
```

Need to pull in the new cmdgw commons library.

--- a/eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml
+++ b/eap6-support/hawkular-javaagent-wildfly-feature-pack-eap6/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml
@@ -562,7 +562,6 @@ resource-type-set-dmr:
     operation-dmr:
     - name: JDR
       internal-name: generate-jdr-report
-      modifies:      true
       path:          /subsystem=jdr
     - name: Reload
       internal-name: reload
@@ -750,6 +749,10 @@ resource-type-set-dmr:
     - name: Base Directory
       attribute: base-dir
       path:      /core-service=server-environment
+    operation-dmr:
+    - name: JDR
+      internal-name: generate-jdr-report
+      path:          /subsystem=jdr
 
   - name: Domain WildFly Server Controller
     resource-name-template: "Domain WildFly Server Controller [%-]"

--- a/eap6-support/hawkular-wildfly-agent-feature-pack-eap6/src/main/resources/subsystem-templates/hawkular-wildfly-agent.xml
+++ b/eap6-support/hawkular-wildfly-agent-feature-pack-eap6/src/main/resources/subsystem-templates/hawkular-wildfly-agent.xml
@@ -732,6 +732,7 @@
         <resource-config-dmr name="Base Directory"
                              path="/core-service=server-environment"
                              attribute="base-dir"/>
+        <operation-dmr name="JDR" internal-name="generate-jdr-report" path="/subsystem=jdr" />
       </resource-type-dmr>
 
       <resource-type-dmr name="Domain WildFly Server Controller"

--- a/hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/cmd/ExportJdrCommand.java
+++ b/hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/cmd/ExportJdrCommand.java
@@ -67,6 +67,7 @@ public class ExportJdrCommand extends AbstractDMRResourcePathCommand<ExportJdrRe
 
         CanonicalPath canonicalPath = CanonicalPath.fromString(request.getResourcePath());
         String resourceId = canonicalPath.ids().getResourcePath().getSegment().getElementId();
+        boolean deleteImmediately = request.isDeleteImmediately();
 
         ResourceManager<DMRNodeLocation> resourceManager = endpointService.getResourceManager();
         Resource<DMRNodeLocation> resource = resourceManager.getResource(new ID(resourceId));
@@ -118,6 +119,17 @@ public class ExportJdrCommand extends AbstractDMRResourcePathCommand<ExportJdrRe
         File reportFile = new File(reportLocation);
         InputStream reportInputStream = new FileInputStream(reportFile);
         binaryData = new BinaryData(null, reportInputStream);
+
+        if (deleteImmediately) {
+            log.tracef("Operation [%s] is going to delete [%s] file after completed.",
+                    requestedOpName, reportFile.getName());
+
+            binaryData.setOnCloseAction(() -> {
+                reportFile.delete();
+                log.tracef("File [%s] is deleted for operation [%s].",
+                        reportFile.getName(), requestedOpName);
+            });
+        }
 
         response.setStatus(ResponseStatus.OK);
         response.setFileName(reportFile.getName());

--- a/hawkular-javaagent-wildfly-feature-pack/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml
+++ b/hawkular-javaagent-wildfly-feature-pack/src/main/resources/featurepack/content/standalone/configuration/hawkular-javaagent-config.yaml
@@ -598,7 +598,6 @@ resource-type-set-dmr:
     operation-dmr:
     - name: JDR
       internal-name: generate-jdr-report
-      modifies:      true
       path:          /subsystem=jdr
     - name: Reload
       internal-name: reload
@@ -885,6 +884,10 @@ resource-type-set-dmr:
     - name: Base Directory
       attribute: base-dir
       path:      /core-service=server-environment
+    operation-dmr:
+    - name: JDR
+      internal-name: generate-jdr-report
+      path:          /subsystem=jdr
 
   - name: Domain WildFly Server Controller
     resource-name-template: "Domain WildFly Server Controller [%-]"

--- a/hawkular-wildfly-agent-feature-pack/src/main/resources/subsystem-templates/hawkular-wildfly-agent.xml
+++ b/hawkular-wildfly-agent-feature-pack/src/main/resources/subsystem-templates/hawkular-wildfly-agent.xml
@@ -801,6 +801,7 @@
         <resource-config-dmr name="Base Directory"
                              path="/core-service=server-environment"
                              attribute="base-dir"/>
+        <operation-dmr name="JDR" internal-name="generate-jdr-report" path="/subsystem=jdr" />
       </resource-type-dmr>
 
       <resource-type-dmr name="Domain WildFly Server Controller"

--- a/pom.xml
+++ b/pom.xml
@@ -72,13 +72,12 @@
     <version.com.samskivert.jmustache>1.11</version.com.samskivert.jmustache>
     <version.io.dropwizard.metrics>3.1.2</version.io.dropwizard.metrics>
     <version.org.hamcrest>1.3</version.org.hamcrest>
-    <version.org.hawkular.commons>0.9.6.Final-SRC-revision-47755c3d21f2ee5f8b7b31b11d7a76e94f911f2c</version.org.hawkular.commons>
+    <version.org.hawkular.commons>0.9.6.Final-SRC-revision-13817b4f13847881159b9c60c1612de5ee450f83</version.org.hawkular.commons>
     <version.org.hawkular.metrics>0.26.1.Final</version.org.hawkular.metrics>
 
     <version.org.jboss.aesh>0.66.7</version.org.jboss.aesh>
     <version.org.jgrapht>0.9.1</version.org.jgrapht>
     <version.org.jolokia>1.3.5</version.org.jolokia>
-
   </properties>
 
   <dependencyManagement>

